### PR TITLE
Add lro enum

### PR DESCRIPTION
**Pull request by m-nash**
# Contributing to the Azure SDK

Adding a new LRO enum to indicate if the method should wait for full completion or return after it has started.

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/core/Azure.Core/src/WaitUntil.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+namespace Azure
+{
+    /// <summary>
+    /// Indicates whether the invocation of a long running operation should return once it has
+    /// started or wait for the operation to fully complete before returning.
+    /// </summary>
+    public enum WaitUntil
+    {
+        /// <summary>
+        /// Indicates the method should wait until the operation fully completes.
```
Comment (KrzysztofCwalina) [inline sdk/core/Azure.Core/src/WaitUntil.cs:13]: Should we add "server" i.e. what until the server operation ...

**Review by KrzysztofCwalina — APPROVED** [review]

File: sdk/core/Azure.Core/src/WaitUntil.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure
```
Comment (annelo-msft) [inline sdk/core/Azure.Core/src/WaitUntil.cs:4]: Curious: is it in the Azure namespace so it sits alongside `Operation<T>`?

**Review by annelo-msft — COMMENTED** [review]
We should add tests to illustrate the intended usage of this type.

Are updates needed to any of the `Operation*` types to enforce usage of this type, or is there an issue tracking that work?

File: sdk/core/Azure.Core/src/WaitUntil.cs (hunk 2 above)
Comment (m-nash) [reply sdk/core/Azure.Core/src/WaitUntil.cs:4]: that was the idea yes

**Comment by m-nash** [comment]
> We should add tests to illustrate the intended usage of this type.
> 
> Are updates needed to any of the `Operation*` types to enforce usage of this type, or is there an issue tracking that work?

No real way to enforce this, you can easily use an `Operation` object as a return for a method that does not have this parameter.  This would be the new standard going forward for LRO methods however which are initially being enforced only in generated clients like LLC and Mgmt

**Review by christothes — APPROVED** [review]

**Comment by annelo-msft** [comment]
Looks like we have this issue tracking the work I mentioned - I just added the Azure.Core label so it doesn't get lost.

https://github.com/Azure/azure-sdk-for-net/issues/25716

**Pull request merged**